File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/MedicinalProductPharmaceutical.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.Resource;

public class MedicinalProductPharmaceutical : DomainResource, IResource
{
    public Base.ResourceReference[]? Device { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ResourceReference[]? Ingredient { get; set; }
    public MedicinalProductPharmaceutical_Characteristics[]? Characteristics { get; set; }
    public Base.CodeableConcept? UnitOfPresentation { get; set; }
    public required Base.CodeableConcept AdministrableDoseForm { get; set; }
    public required MedicinalProductPharmaceutical_RouteOfAdministration[] RouteOfAdministration { get; set; }

    public class MedicinalProductPharmaceutical_Characteristics : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public Base.CodeableConcept? Status { get; set; }
    }

    public class MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies_WithdrawalPeriod : BackboneElement
    {
        public required Base.Quantity Value { get; set; }
        public required Base.CodeableConcept Tissue { get; set; }
        public string? SupportingInformation { get; set; }
    }

    public class MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies_WithdrawalPeriod[]? WithdrawalPeriod { get; set; }
    }

    public class MedicinalProductPharmaceutical_RouteOfAdministration : BackboneElement
    {
        public required Base.CodeableConcept Code { get; set; }
        public Base.Quantity? FirstDose { get; set; }
        public Base.Quantity? MaxDosePerDay { get; set; }
        public Base.Quantity? MaxSingleDose { get; set; }
        public MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies[]? TargetSpecies { get; set; }
        public string? MaxTreatmentPeriod { get; set; }
        public Base.Ratio? MaxDosePerTreatmentPeriod { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/MedicinalProductPharmaceutical.py
from typing import Optional, List
from base import CodeableConcept
from base import Quantity
from base import Duration
from base import DomainResource
from base import Ratio
from base import Reference
from base import Identifier
from base import BackboneElement

class MedicinalProductPharmaceutical_Characteristics(BackboneElement):
    code: CodeableConcept
    status: Optional[CodeableConcept] = None

class MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies_WithdrawalPeriod(BackboneElement):
    value: Quantity
    tissue: CodeableConcept
    supporting_information: Optional[str] = None

class MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies(BackboneElement):
    code: CodeableConcept
    withdrawal_period: Optional[List[MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies_WithdrawalPeriod]] = None

class MedicinalProductPharmaceutical_RouteOfAdministration(BackboneElement):
    code: CodeableConcept
    first_dose: Optional[Quantity] = None
    max_dose_per_day: Optional[Quantity] = None
    max_single_dose: Optional[Quantity] = None
    target_species: Optional[List[MedicinalProductPharmaceutical_RouteOfAdministration_TargetSpecies]] = None
    max_treatment_period: Optional[Duration] = None
    max_dose_per_treatment_period: Optional[Ratio] = None

class MedicinalProductPharmaceutical(DomainResource):
    device: Optional[List[Reference]] = None
    identifier: Optional[List[Identifier]] = None
    ingredient: Optional[List[Reference]] = None
    characteristics: Optional[List[MedicinalProductPharmaceutical_Characteristics]] = None
    unit_of_presentation: Optional[CodeableConcept] = None
    administrable_dose_form: CodeableConcept
    route_of_administration: list[MedicinalProductPharmaceutical_RouteOfAdministration] = []